content/doc/book/managing/cli.adoc

```diff
@@ -0,0 +1,285 @@
+a build will only run if there has been an SCM change.
+ JOB : Name of the job to build
+ -c  : Check for SCM changes before starting the build, and if there's no
+       change, exit without doing a build
+ -f  : Follow the build progress. Like -s only interrupts are not passed
+       through to the build.
+ -p  : Specify the build parameters in the key=value format.
+ -s  : Wait until the completion/abortion of the command. Interrupts are passed
+       through to the build.
+ -v  : Prints out the console output of the build. Use with -s
+ -w  : Wait until the start of the command
+% ssh -l kohsuke -p 53801 localhost build build-all-software -f -v
+Started build-all-software #1
+Started from command line by admin
+Building in workspace /tmp/jenkins/workspace/build-all-software
+[build-all-software] $ /bin/sh -xe /tmp/hudson1100603797526301795.sh
++ echo hello world
+hello world
+Finished: SUCCESS
+Completed build-all-software #1 : SUCCESS
+%
+----
+
+==== console
+
+Similarly useful is the `console` command which retrieves the console output
+for the specified build or Pipeline run. When no build number is provided, the
+`console` command will output the last completed build's console output.
+
+[source]
+----
+% ssh -l kohsuke -p 53801 localhost help console
+
+java -jar jenkins-cli.jar console JOB [BUILD] [-f] [-n N]
+Produces the console output of a specific build to stdout, as if you are doing 'cat build.log'
+ JOB   : Name of the job
+ BUILD : Build number or permalink to point to the build. Defaults to the last
+         build
+ -f    : If the build is in progress, stay around and append console output as
+         it comes, like 'tail -f'
+ -n N  : Display the last N lines
+% ssh -l kohsuke -p 53801 localhost console build-all-software
+Started from command line by kohsuke
+Building in workspace /tmp/jenkins/workspace/build-all-software
+[build-all-software] $ /bin/sh -xe /tmp/hudson1100603797526301795.sh
++ echo hello world
+yes
+Finished: SUCCESS
+%
+----
+
+==== who-am-i
+
+The `who-am-i` command is helpful for listing the current user's credentials
+and permissions available to the user. This can be useful when debugging the
+absence of CLI commands based on the lack of certain permissions.
+
+[source]
+----
+
+% ssh -l kohsuke -p 53801 localhost help who-am-i
+
+java -jar jenkins-cli.jar who-am-i
+Reports your credential and permissions.
+% ssh -l kohsuke -p 53801 localhost who-am-i
+Authenticated as: kohsuke
+Authorities:
+  authenticated
+%
+----
+
+
+== Using the CLI client
+
+
+While the SSH-based CLI is preferred, there may be situations where the CLI
+client is a better fit. For example, the default transport for the CLI client
+is HTTP which means no additional ports must be opened in a firewall for its
+use.
+
+
+=== Downloading the client
+
+The CLI client can be downloaded directly from a Jenkins master at the URL
+`/jnlpJars/jenkins-cli.jar`, in effect  `https://JENKINS_URL/jnlpJars/jenkins-cli.jar`
```

Comment:
It's not enabled by default, but I have been using it more as a subtle nod that their JENKINS_URL should use HTTPs.